### PR TITLE
Add a hidden option for keeping the encryptor running after failure

### PR DESCRIPTION
**Pull request by bburtin**
Add a hidden --no-terminate-encryptor-on-failure option.  When this
option is specified and encryption fails, we terminate the guest
instance but not the encryptor.  This allows a developer to connect to
the encryptor instance to diagnose the problem.

### CODE REVIEW

File: brkt_cli/aws/encrypt_ami.py [hunk 1]
```diff
@@ -1065,26 +1066,27 @@ def encrypt(aws_svc, enc_svc_cls, image_id, encryptor_ami,
         instance_ids = []
         if guest_instance:
             instance_ids.append(guest_instance.id)
-        if encryptor_instance:
+        if encryptor_instance and terminate_encryptor_on_failure:
             instance_ids.append(encryptor_instance.id)
```
Comment (agtucker) [inline brkt_cli/aws/encrypt_ami.py]: Maybe print something like "preserving encryptor instance"?

File: brkt_cli/aws/encrypt_ami_args.py [hunk 2]
```diff
@@ -124,10 +124,21 @@ def setup_encrypt_ami_args(parser):
...
         dest='save_encryptor_logs',
         action='store_true',
         help=argparse.SUPPRESS,
         default=False
     )
+
+    # Optional argument for development: if encryption fails, keep the
+    # encryptor running so that we can debug it.
+    parser.add_argument(
+        '--no-terminate-encryptor-on-failure',
```
Comment (agtucker) [inline brkt_cli/aws/encrypt_ami_args.py]: Just noticed that the GCE code has a --keep-encryptor option that does the same thing - use the same argument name?

Comment (bburtin) [inline brkt_cli/aws/encrypt_ami_args.py]: I think it's different.  From looking at the code, it looks like that flag controls whether we delete the metavisor image in the GCP project, not the encryptor instance.

File: brkt_cli/aws/encrypt_ami.py (hunk 1 above)
Comment (bburtin) [inline brkt_cli/aws/encrypt_ami.py]: Good idea.

File: brkt_cli/aws/encrypt_ami_args.py (hunk 2 above)
Comment (agtucker) [inline brkt_cli/aws/encrypt_ami_args.py]: Yeah, I think you're right. Then what about GCE implementation of the "no terminate" functionality? Should I file a separate bug for that?

Comment (bburtin) [inline brkt_cli/aws/encrypt_ami_args.py]: Yeah, I guess so.  I know the AWS code a lot better, so I'm able to bang this out pretty quickly.  Go ahead and file the GCE bug and either Bobby or I will look at it separately.

File: brkt_cli/aws/encrypt_ami.py [hunk 3]
```diff
@@ -1065,26 +1066,39 @@ def encrypt(aws_svc, enc_svc_cls, image_id, encryptor_ami,
         instance_ids = []
         if guest_instance:
             instance_ids.append(guest_instance.id)
-        if encryptor_instance:
+
+        terminate_encryptor = (
+            ami or
+            (encryptor_instance and terminate_encryptor_on_failure)
```
Comment (agtucker) [inline brkt_cli/aws/encrypt_ami.py]: Should this be:
    (encryptor_instance and (ami or terminate_encryptor_on_failure))
I.e. don't terminate the encryptor unless there is one. Though I guess ami can't be set unless encryptor_instance is also set, so it works out to be the same. LGTM either way.

Comment (bburtin) [inline brkt_cli/aws/encrypt_ami.py]: Yeah, reads better, given that the logic acts as a null check.  I'll fix it.

**Pull request merged**